script/tool/lib/src/update_dependency_command.dart

```diff
@@ -233,59 +250,110 @@ ${response.httpResponse.body}
-            .hasMatch(gradleWrapperPropertiesContents)) {
-          return PackageResult.fail(<String>[
-            'Unable to find a "distributionUrl" entry to update for ${package.displayName}.'
-          ]);
-        }
-
-        print(
-            '${indentation}Updating ${getRelativePosixPath(example.directory, from: package.directory)} to "$_targetVersion"');
-        final String newGradleWrapperPropertiesContents =
-            gradleWrapperPropertiesContents.replaceFirst(
-                validGradleDistributionUrl,
-                'distributionUrl=https\\://services.gradle.org/distributions/gradle-$_targetVersion-all.zip');
         // TODO(camsim99): Validate current AGP version against target Gradle
         // version: https://github.com/flutter/flutter/issues/133887.
-        gradleWrapperPropertiesFile
-            .writeAsStringSync(newGradleWrapperPropertiesContents);
+        newDependencyVersionEntry =
+            'distributionUrl=https\\://services.gradle.org/distributions/gradle-$_targetVersion-all.zip';
+      } else if (_targetAndroidDependency ==
+          _AndroidDepdencyType.compileSdkForExamples) {
+        fileToUpdate =
+            androidDirectory.childDirectory('app').childFile('build.gradle');
+        dependencyVersionPattern = RegExp(
+            r'(compileSdk|compileSdkVersion) (\d{1,2}|flutter.compileSdkVersion)');
+        newDependencyVersionEntry = 'compileSdk $_targetVersion';
       }
-      return updateRanForExamples
-          ? PackageResult.success()
-          : PackageResult.skip('No example apps run on Android.');
+
+      final String oldFileToUpdateContents = fileToUpdate!.readAsStringSync();
+
+      if (!dependencyVersionPattern!.hasMatch(oldFileToUpdateContents)) {
+        return PackageResult.fail(<String>[
+          'Unable to find a $_targetAndroidDependency version entry to update for ${example.displayName}.'
+        ]);
+      }
+
+      print(
+          '${indentation}Updating ${getRelativePosixPath(example.directory, from: package.directory)} to "$_targetVersion"');
+      final String newGradleWrapperPropertiesContents = oldFileToUpdateContents
+          .replaceFirst(dependencyVersionPattern, newDependencyVersionEntry!);
+
+      fileToUpdate.writeAsStringSync(newGradleWrapperPropertiesContents);
     }
-    return PackageResult.fail(<String>[
-      'Target Android dependency $_androidDependency is unrecognized.'
-    ]);
+    return updateRanForExamples
+        ? PackageResult.success()
+        : PackageResult.skip('No example apps run on Android.');
+  }
+
+  Future<PackageResult> _runForCompileSdkVersion(
+      RepositoryPackage package) async {
+    if (!package.platformDirectory(FlutterPlatform.android).existsSync()) {
+      return PackageResult.skip(
+          'Package ${package.displayName} does not run on Android.');
+    } else if (package.getEnclosingPackage() != null) {
+      // We skip examples for this command.
```

Comment:
Would appreciate if someone can sanity-check me on whether or not this is a valid way to ensure we don't run on examples running on Android.

Reply:
There are edge cases where it probably isn't exactly what you want; https://github.com/flutter/packages/tree/main/packages/pigeon/platform_tests/test_plugin comes to mind (this is a plugin which in inside a package, but which we probably want automagic updates to work on).

If you want it to specifically apply to examples, you could make a `RepositoryPackage` `isExample` method that calls `getEnclosingPackage`, then calls `getExamples` on that, and then checks if any of them is the package you started with.